### PR TITLE
Initial FortiOS support

### PR DESCRIPTION
**Pull request by Schamper**
This PR adds initial support for FortiOS images (currently tested with the VMs of FortiGate and FortiAnalyzer. The goal is to have some initial support available at first, and expand later on as more time for research becomes available.

For example, we also have some parsers for various log files, but they require some cleaning up as well as needing to figure out how feasible it is to parse those straight from a FortiOS image.

### CODE REVIEW

File: dissect/target/plugins/os/unix/linux/fortios/_os.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    def os(self) -> str:
+        return OperatingSystem.FORTIGATE.value
+
+    @export(record=UnixUserRecord)
+    def users(self) -> Iterator[UnixUserRecord]:
+        # TODO: Add FortiOS specific users
+        return super().users()
+
+
+def repair_lzma_stream(fh: BinaryIO) -> BinaryIO:
```
Comment (sud0woodo) [inline dissect/target/plugins/os/unix/linux/fortios/_os.py:113]: <3

**Comment by Schamper** [comment]
Now depends on https://github.com/fox-it/dissect.util/pull/39

File: dissect/target/plugin.py [hunk 2]
```diff
@@ -683,10 +683,14 @@ def _walk(
...
-        _get_plugins(),
-        special_keys=special_keys,
-        only_special_keys=only_special_keys,
+    yield from sorted(
+        _walk(
+            _get_plugins(),
+            special_keys=special_keys,
+            only_special_keys=only_special_keys,
+        ),
+        key=lambda el: len(el["module"]),
```
Comment (pyrco) [inline dissect/target/plugin.py:692]:
```suggestion
        key=lambda plugin: len(plugin["module"]),
```
or `plugin_desc` or something

I noticed I didn't add any return type to `_walk()` which doesn't help either.

**Review by pyrco — CHANGES_REQUESTED** [review]

File: dissect/target/plugins/os/unix/linux/fortios/_os.py [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+
+        return result
+
+    @export(property=True)
+    def dns(self) -> list[str]:
+        """Return configured WAN DNS servers."""
+        entries = []
+        for _, entry in self._config["global-config"]["system"]["dns"].items():
+            entries.append(entry[0])
+        return entries
```
Comment (pyrco) [inline dissect/target/plugins/os/unix/linux/fortios/_os.py:163]: This is not an OSPlugin method and should be put somewhere else.

Comment (Schamper) [reply dissect/target/plugins/os/unix/linux/fortios/_os.py:163]: This currently overrides from `LinuxPlugin`, so I propose to leave it be for now, until we do a big OSPlugin cleanup.

Comment (pyrco) [reply dissect/target/plugins/os/unix/linux/fortios/_os.py:163]: I see, that is nasty.

**Review by pyrco — APPROVED** [review]

**Review by pyrco — CHANGES_REQUESTED** [review]
There is one testcase that fails

**Comment by Schamper** [comment]
> There is one testcase that fails

Likely because this PR now depends on https://github.com/fox-it/dissect.util/pull/39. I moved the XZ repair code into dissect.util.

**Review by pyrco — APPROVED** [review]

**Pull request merged**